core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -302,6 +306,9 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
       nextQuery = nextQuery)
   }
 
+  // Note: very sneaky that this looks like a method but actually is a lambda that keeps state across invocations and
+  // also that it is called deserializeAndAddOffset but actually also filters out duplicates, can we make it something
+  // more obviously stateful while still testable in isolation (a Flow[SerializedJournalRow, EventEnvelope] perhaps)?
```

Comment:
Sounds good to improve that. I think it's a rather exact copy from spanner plugin. It's used in statefulMapConcat so it's not wrong that it has state.

Reply:
a quick improvement could be to have `def ..: Flow` and put the statefulMapConcat here together with the function